how-to/laravel.md

```diff
@@ -0,0 +1,653 @@
+    }
+}
+```
+
+- after
+```php
+<?php
+
+namespace Database\Factories;
+
+use Illuminate\Database\Eloquent\Factories\Factory;
+
+/**
+ * @extends \Illuminate\Database\Eloquent\Factories\Factory<\App\Models\Post>
+ */
+class PostFactory extends Factory
+{
+    /**
+     * Define the model's default state.
+     *
+     * @return array<string, mixed>
+     */
+    public function definition(): array
+    {
+        return [
+            'title' => fake()->realText(20),
+            'body' => fake()->realText(200),
+        ];
+    }
+}
+```
+
+
+**## Database Seeder**
+
+Now you need to tell the database seeder class to use the factory we’ve created. We will modify the seeder to create 200,000 dummy blog posts:
+
+- before
+```php
+<?php
+
+namespace Database\Seeders;
+
+// use Illuminate\Database\Console\Seeds\WithoutModelEvents;
+use Illuminate\Database\Seeder;
+
+class DatabaseSeeder extends Seeder
+{
+    /**
+     * Seed the application's database.
+     */
+    public function run(): void
+    {
+        // \App\Models\User::factory(10)->create();
+
+        // \App\Models\User::factory()->create([
+        //     'name' => 'Test User',
+        //     'email' => 'test@example.com',
+        // ]);
+    }
+}
+```
+
+- after
+```php
+<?php
+
+namespace Database\Seeders;
+
+// use Illuminate\Database\Console\Seeds\WithoutModelEvents;
+use Illuminate\Database\Seeder;
+
+class DatabaseSeeder extends Seeder
+{
+    /**
+     * Seed the application's database.
+     */
+    public function run(): void
+    {
+        // \App\Models\User::factory(10)->create();
+
+        // \App\Models\User::factory()->create([
+        //     'name' => 'Test User',
+        //     'email' => 'test@example.com',
+        // ]);
+
+		  echo "Now you need to wait for few minitues \n";
+
+        for($i =0; $i<2000; $i++)
+        {
+            \App\Models\Post::factory()->count(100)->create();
+        }
+    }
+}
+```
+
+**## Run migration and seed sample data**
+
+Following command will run the migration to PostgreSQL then seed 
+ fake data to it.
+
+```shell
+php arrtisan migrate
+
+php artisan db:seed
+```
+
+
+## Let’s test it
+First, you need to build a css. Just run the following command.
+```shell
+npm run build
+```
+
+Then start the laravel by running follwoing command.
+```shell
+php artisan serve
+```
+
+Now you can access the application in your web browser at: [http://localhost:8000](http://localhost:8000)
+
+<img width="1024" alt="laravel_sample" src="https://github.com/askdkc/pgroonga.github.io/assets/7894265/f8557078-85a7-474f-95f3-3c405b1b6ffa">
+
+
+## Search Function
+Good thing about `PGroonga` is, unlike popular RDB like MySQL and PostgreSQL which only can use indexes for prefix match search, it can also use indexes for Full-text partial match search! Not only that, you can get AND search and OR search for free! Just type your keyword separate with space, you get AND search. If you separate words with OR (capital letter), you get OR search.
```

Comment:
```suggestion
Good thing about `PGroonga` is, unlike popular RDB like MySQL and PostgreSQL which only can use indexes for prefix match search, it can also use indexes for full-text partial match search! Not only that, you can get AND search and OR search for free! Just type your keyword separate with space, you get AND search. If you separate words with OR (capital letter), you get OR search.
```

Reply:
ditto.